mixer/tools/adapterlinter/main.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"os"
+	"sort"
+	"strings"
+)
+
+var exitCode int
+
+var invalidImportPaths = map[string]string{
+	"log": "\"log\" import is not recommended; Adapters must instead use env.Logger for logging during execution. " +
+		"This logger understands about which adapter is running and routes the data to the place where the operator " +
+		"wants to see it.",
+	"github.com/golang/glog": "\"github.com/golang/glog\" import is not recommended; Adapters must instead use env.Logger for logging during execution. " +
+		"This logger understands about which adapter is running and routes the data to the place where the operator " +
+		"wants to see it.",
+}
+
+func main() {
+	flag.Parse()
+	var reports []string
+	if flag.NArg() == 0 {
```

Comment:
Might be nice to support the same kind of arguments as other go tools in terms of ./... kind of thing.